consensus/istanbul/core/core.go

```diff
@@ -376,26 +376,28 @@ func (c *core) waitForDesiredRound(r *big.Int) {
 
 func (c *core) resetRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, roundChange bool) {
 	// TODO(Joshua): Include desired round here.
-	if c.current != nil {
-		if roundChange {
-			c.current = newRoundState(view, validatorSet, c.current.PendingRequest(), c.current.PreparedCertificate(), c.current.ParentCommits())
-		} else {
-			lastSubject, err := c.backend.LastSubject()
-			if err != nil && c.current.Proposal() != nil && c.current.Proposal().Hash() == lastSubject.Digest && c.current.Round().Cmp(lastSubject.View.Round) == 0 {
-				// When changing sequences, if our current Commit messages match the latest block in the chain
-				// (i.e. they're for the same block hash and round), we use this sequence's commits as the ParentCommits field
-				// in the next round.
-				c.current = newRoundState(view, validatorSet, nil, istanbul.EmptyPreparedCertificate(), c.current.Commits())
-			} else {
-				headBlock := c.backend.GetCurrentHeadBlock()
-				// Otherwise, we will initialize an empty ParentCommits field with the validator set of the last proposal.
-				c.current = newRoundState(view, validatorSet, nil, istanbul.EmptyPreparedCertificate(), newMessageSet(c.backend.ParentBlockValidators(headBlock)))
-			}
-		}
-	} else {
+	if c.current == nil {
 		// When the current round is nil, we must start with the current validator set in the parent commits
 		// either `validatorSet` or `backend.Validators(lastProposal)` works here
-		c.current = newRoundState(view, validatorSet, nil, istanbul.EmptyPreparedCertificate(), newMessageSet(validatorSet))
+		c.current = newRoundState(view, validatorSet)
+	} else if roundChange {
+		c.current.StartNewRound(view.Round, validatorSet)
```

Comment:
Previously, we would have passed `c.current.PendingRequest()`  to the round state, but it now looks like we always set it to nil after a round change?

Reply:
before we would be overriding c.current with a new instance. now we mutate it, changing only the necessary fields.

So `StartNewRound` should behave exactly as calling the old `newRoundState` with the old parameters. Since the only new information is the view and validatorSet, the other parameters are not needs, and are not changed on StartNewRound